docs/modules/ROOT/pages/spring-cloud-commons/common-abstractions.adoc

```diff
@@ -167,13 +167,140 @@ CAUTION: A `RestTemplate` bean is no longer created through auto-configuration.
+
+TIP: If you see errors such as `java.lang.IllegalArgumentException: Can not set org.springframework.web.client.RestTemplate field com.my.app.Foo.restTemplate to com.sun.proxy.$Proxy89`, try injecting `RestOperations` or setting `spring.aop.proxyTargetClass=true`.
+
+[[rest-client-loadbalancer-client]]
+== Spring `RestClient` as a LoadBalancer Client
+
+You can configure a `RestClient` to use a Load-balancer client.
+To create a load-balanced `RestClient`, create a `RestClient.Builder` `@Bean` and use the `@LoadBalanced` qualifier, as the following example shows:
+
+[source,java,indent=0]
+----
+@Configuration
+public class MyConfiguration {
+
+    @LoadBalanced
+    @Bean
+    RestClient.Builder restClientBuilder() {
+        return RestClient.builder();
+    }
+}
+
+public class MyClass {
+
+    @Autowired
+    private RestClient.Builder restClientBuilder;
+
+    public String doOtherStuff() {
+        return restClientBuilder.build().get().uri(URI.create("http://stores/stores")).retrieve().body(String.class);
+    }
+}
+----
+
+The URI needs to use a virtual host name (that is, a service name, not a host name).
+The `BlockingLoadBalancerClient` is used to create a full physical address.
+
+IMPORTANT: To use a load-balanced `RestClient`, you need to have a Spring LoadBalancer implementation in your classpath.
+Add xref:spring-cloud-commons/loadbalancer.adoc#spring-cloud-loadbalancer-starter[Spring Cloud LoadBalancer starter] to your project in order to use it.
+
+[[multiple-restclient-objects]]
+=== Multiple `RestClient.Builder` Objects
+
+If you want a `RestClient.Builder` that is not load-balanced, create a `RestClient.Builder` bean and inject it.
+To access the load-balanced `RestClient`, use the `@LoadBalanced` qualifier when you create your `@Bean`, as the following example shows:
+
+[source,java,indent=0]
+----
+@Configuration
+public class MyConfiguration {
+
+    @LoadBalanced
+    @Bean
+    RestClient.Builder loadBalanced() {
+        return RestClient.builder();
+    }
+
+    @Primary
+    @Bean
+    RestClient.Builder restClientBuilder() {
+        return WebClient.builder();
+    }
+}
+
+public class MyClass {
+    @Autowired
+    private RestClient.Builder restClientBuilder;
+
+    @Autowired
+    @LoadBalanced
+    private RestClient.Builder loadBalanced;
+
+    public String doOtherStuff() {
+        return restClientBuilder.build().get().uri("http://stores/stores")
+        				.retrieve().body(String.class);
+    }
+
+    public String doStuff() {
+        return restClientBuilder.build().get().uri("http://example.com")
+        				.retrieve().body(String.class);
+    }
+}
+----
+
+IMPORTANT: Notice the use of the `@Primary` annotation on the plain `RestTemplate` declaration in the preceding example to disambiguate the unqualified `@Autowired` injection.
+
 [[webclinet-loadbalancer-client]]
-== Spring WebClient as a Load Balancer Client
+== Spring `WebClient` as a LoadBalancer Client
```

Comment:
Remove the ticks around `WebClient`. The code font does not render correctly in headings.